smithy-aws-protocol-tests/model/awsJson1_0/empty-input-output.smithy

```diff
@@ -16,28 +16,97 @@ operation NoInputAndNoOutput {}
 
 apply NoInputAndNoOutput @httpRequestTests([
     {
-        id: "AwsJson10NoInputAndNoOutput",
-        documentation: "No input serializes no payload",
+        id: "AwsJson10MustAlwaysSendEmptyJsonPayload",
+        documentation: """
+                Clients must always send an empty JSON object payload for
+                operations with no input (that is, `{}`). While AWS service
+                implementations support requests with no payload or requests
+                that send `{}`, always sending `{}` from the client is
+                preferred for forward compatibility in case input is ever
+                added to an operation.""",
         protocol: awsJson1_0,
         method: "POST",
         headers: {
             "Content-Type": "application/x-amz-json-1.0",
             "X-Amz-Target": "JsonRpc10.NoInputAndNoOutput",
         },
         uri: "/",
-    }
+        body: "{}",
+        bodyMediaType: "application/json"
+    },
+    {
+        id: "AwsJson10ServiceSupportsNoPayloadForNoInput",
+        documentation: """
+                Service implementations must support no payload or an empty
+                object payload for operations that define no input. However,
+                despite the lack of a payload, a Content-Type header is still
+                required in order for the service to properly detect the
+                protocol.""",
+        protocol: awsJson1_0,
+        method: "POST",
+        headers: {
+            "Content-Type": "application/x-amz-json-1.0",
+            "X-Amz-Target": "JsonRpc10.NoInputAndNoOutput",
+        },
+        uri: "/",
+        body: "",
+        tags: ["server-only"]
```

Comment:
server only vs. client only seems fundamental enough that it may be worth modeling

Reply:
It does feel that way. Let's do that in a follow up PR though since these tags are already in use.